### ISSUE TITLE
Zoom?

### ISSUE THREAD

**Issue by excitedbox** [umaranis/MindMate#67]
Am I missing it or is there no zoom?

Also a universally editable export format such as XML or Json is an absolute must have these days.

Another nice feature would be an easier way to control which side a branch is on. I think more visualization options in general would be awesome.

PS. Ant vision G6 is an amazing visualization engine with thousands of features, if you ever decide to make another version in the future you can save yourself 95% of the work by using that as a base. [Antv G6](https://g6.antv.vision/en)

**Comment by umaranis** [comment]
Hi @excitedbox ,

There is no zoom feature as yet.

**Comment by umaranis** [comment]
I had a quick look at Antv G6. It seems mostly about visualization while Mind Mate is about taking notes.

**Comment by excitedbox** [comment]
Yes, G6 is a visualization framework used to make Xml/Json/DB driven graphs it includes all the features needed for mind mapping and one of their demos is a mind map editor. It is used with Angular or React to make web dashboards but you could use Electron to make a Desktop offline app as well. You really only need to make a CSS layout and place the buttons. All the editing interactive stuff is already in the API. 

If you look at the bottom of their website there are several different versions used for different things. G6 and x6 would probably be the most useful. The English documentation is pretty good as well.

I think if I really tried it would take about 2 days to recreate all the features of MindMate using this because it only needs a UI to tie all the features together. I am looking at using X6 to make a node based programing IDE because it really has everything included.

This is the mindmap demo
https://g6.antv.vision/en/examples/tree/mindmap#hMindmap

If I was to use it to make a MindMate clone I would make a Template for the toolbars, viewer and the UI and use the JS API to update some Json or XML files. Those then get displayed in the viewer. I don't have much experience with Electron but I know it is used to run React and Angular apps as offline programs very easily. It creates a local http server which serves the app locally and wraps the page in a custom browser window.

**Comment by excitedbox** [comment]
PS. this is an editor demo they made using g6 It can also edit mind maps. https://github.com/antvis/g6-editor

**Comment by umaranis** [comment]
Hi @excitedbox ,

MindMate has a number of advanced features that would require significant effort to re-develop. I would name just a few:
- ability to save images (one advanced aspect of it: user can copy a whole webpage into it, it will not only save the text but also download and save the images locally)
- unlimited undo/redo
- task management
- clipboard integration


I like the ideas you shared, but that would be a different tool serving different needs. MindMate is different in the sense that it is very lightweight (uses just a few MBs of RAM) and is much faster (being native). These features make sense, for instance, if you are using it as a research assistant tool where you may have to bring in stuff from the local drive and the web. 

Thanks for your input. I would look forward to your node based IDE :)

**Comment by spongioblast** [comment]
Any possibility to get zoom working? That would really help with big mindmaps.
